### PR TITLE
Implement onWillSaveTextDocument plugin api event handler. 

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### Related issue:
[Implement onWillSaveTextDocument event](https://github.com/theia-ide/theia/issues/4009)

### Requested changes:
-  Implement onWillSaveTextDocument plugin api event handler. 
-  Align onWillSaveLoop time out value with VSCode(1500ms)

Signed-off-by: Oleksandr Andriienko <oandriie@redhat.com>

### CODE REVIEW

File: packages/plugin-ext/src/main/browser/documents-main.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class DocumentsMainImpl implements DocumentsMain {
         modelService: EditorModelService,
         rpc: RPCProtocol,
         private editorManger: EditorManager,
-        private openerService: OpenerService
+        private openerService: OpenerService,
```
Comment (olexii4) [inline packages/plugin-ext/src/main/browser/documents-main.ts]: final commas are allowed but  I think we shouldn't add it in existing code
WDYT?

Comment (AndrienkoAleksandr) [reply packages/plugin-ext/src/main/browser/documents-main.ts]: OK

Comment (benoitf) [reply packages/plugin-ext/src/main/browser/documents-main.ts]: it avoids to edit the previous lines when adding new code. For example the previous line about editormanager was not modified when adding this line so to me I find it very nice.

**Comment by AndrienkoAleksandr** [comment]
@benoitf @akosyakov I am going to merge it. Is it ok for you?

**Comment by sunix** [comment]
@benoitf @akosyakov is it fine for you ?

**Pull request merged**